examples/godot_tps_controller_port/src/player.rs

```diff
@@ -0,0 +1,680 @@
+            unsafe { sub_nodes.shoot_particle.assume_safe().set_emitting(true) };
+
+            unsafe { sub_nodes.muzzle_particle.assume_safe().restart() };
+            unsafe { sub_nodes.muzzle_particle.assume_safe().set_emitting(true) };
+            unsafe { sub_nodes.fire_cooldown.assume_safe().start(-1.0) };
+            unsafe { sub_nodes.sound_effect_shoot.assume_safe().play(0.0) };
+            self.add_camera_shake_trauma(sub_nodes.camera_camera, 0.35);
+        }
+    }
+
+    // Not in air or aiming, idle.
+    fn transition_to_idle_state(
+        &mut self,
+        sub_nodes: SubNodes,
+        camera_x: Vector3,
+        camera_z: Vector3,
+        delta: f32,
+    ) {
+        // Convert orientation to quaternions for interpolating rotation.
+        let target = camera_x * self.motion.x + camera_z * self.motion.y;
+        if target.length() > 0.001 {
+            let q_from = self.orientation.basis.to_quat();
+            let q_to = Transform::default()
+                .looking_at(target, Vector3::UP)
+                .basis
+                .to_quat();
+            // nterpolate current rotation with desired one.
+            self.orientation.basis =
+                Basis::from_quat(q_from.slerp(q_to, delta * ROTATION_INTERPOLATE_SPEED));
+        }
+
+        // Aim to zero (no aiming while walking).
+        unsafe {
+            sub_nodes
+                .animation_tree
+                .assume_safe()
+                .set("parameters/aim/add_amount", 0.0)
+        };
+        // Change state to walk.
+        unsafe {
+            sub_nodes
+                .animation_tree
+                .assume_safe()
+                .set("parameters/state/current", 1.0)
+        };
+        // Blend position for walk speed based on motion.
+        unsafe {
+            sub_nodes.animation_tree.assume_safe().set(
+                "parameters/walk/blend_position",
+                Vector2::new(self.motion.length(), 0.0),
+            )
+        };
+
+        self.root_motion = unsafe {
+            sub_nodes
+                .animation_tree
+                .assume_safe()
+                .get_root_motion_transform()
+        };
+    }
+}
+
+// Only __one__ `impl` block can have the `#[methods]` attribute, which
+// will generate code to automatically bind any exported methods to Godot.
+#[methods]
+impl Player {
+    fn new(owner: &KinematicBody) -> Self {
+        Player {
+            // VARIABLES
+            airborne_time: 100.0,
+            orientation: owner.transform(),
+            root_motion: owner.transform(),
+            motion: Vector2::default(),
+            velocity: Vector3::default(),
+
+            aiming: false,
+
+            // If `true`, the aim button was toggled on by a short press (instead of being held down).
+            toggled_aim: false,
+
+            //# The duration the aiming button was held for (in seconds).
+            aiming_timer: 0.0,
+
```

Comment:
These seem to be the same comments as those in the declaration, and should be safe to remove.